backend/models/courseelement.mgmodel.ts

```diff
@@ -0,0 +1,214 @@
+import mongoose, { Schema } from "mongoose";
+import {
+  DisplayElementType,
+  ElementType,
+  HybridElementType,
+  InteractiveElementType,
+} from "../types/courseTypes";
+
+export interface CourseElement {
+  type: ElementType;
+}
+
+const CourseElementSchema: Schema = new Schema({
+  type: {
+    type: String,
+    required: true,
+    enum: [
+      "TextInput",
+      "NumberInput",
+      "CheckboxInput",
+      "MultipleChoice",
+      "Matching",
+      "Table",
+      "RichText",
+      "Image",
+    ],
+  },
+});
+
+export interface RichTextElement extends CourseElement {
+  type: DisplayElementType.RichText;
+  content: string;
+  backgroundColor?: string;
+}
+
+const RichTextElementSchema: Schema = new Schema({
+  content: {
+    type: String,
+    required: true,
+  },
+  backgroundColor: {
+    type: String,
+    required: false,
+  },
+});
+
+export interface ImageElement extends CourseElement {
+  type: DisplayElementType.Image;
+  src: string;
+  alt: string;
+}
+
+const ImageElementSchema: Schema = new Schema({
+  src: {
+    type: String,
+    required: true,
+  },
+  alt: {
+    type: String,
+    required: true,
+  },
+});
+
+export interface TextInputElement extends CourseElement {
+  type: InteractiveElementType.TextInput;
+  correctAnswer?: string;
+}
+
+const TextInputElementSchema: Schema = new Schema({
+  correctAnswer: {
+    type: String,
+    required: false,
+  },
+});
+
+export interface NumberInputElement extends CourseElement {
+  type: InteractiveElementType.NumberInput;
+  correctAnswer?: number;
+  prefixText?: string;
+  suffixText?: string;
+}
+
+const NumberInputElementSchema: Schema = new Schema({
+  correctAnswer: {
+    type: Number,
+    required: false,
+  },
+  prefixText: {
+    type: String,
+    required: false,
+  },
+  suffixText: {
+    type: String,
+    required: false,
+  },
+});
+
+export interface CheckboxInputElement extends CourseElement {
+  type: InteractiveElementType.CheckboxInput;
+  correctAnswer: boolean;
```

Comment:
did u mean to make this field optional because looking at the other elements above, optional ones are marks as such